entrypoint/README.md

```diff
@@ -0,0 +1,145 @@
+# Entrypoint
```

Comment:
This is a fantastic write up! One nit which I do not think is worth looking at in this PR itself is I believe it would be best to have the "what and how to use" up front followed by the "why". I believe more people will be interested in what it is and how to use it before they end up asking what brought us from where we are today to this pattern.